packages/ifpack2/src/Ifpack2_Container.hpp

```diff
@@ -1069,7 +1069,7 @@ void ContainerImpl<MatrixType, LocalScalarType>::DoGSBlock(
     //Use the KokkosSparse internal matrix for low-overhead values/indices access
     //But, can only do this if the matrix is accessible directly from host, since it's not a DualView
     auto localA = this->inputCrsMatrix_->getLocalMatrix();
-    typedef typename decltype(localA)::size_type size_type;
+    using size_type = typename decltype(localA)::size_type;
     const auto& rowmap = localA.graph.row_map;
```

Comment:
1. You're accessing UVM allocations on host.  That's OK, but you may want to fence first to ensure that any device kernels are done before you access the allocations on host.
2. Be careful with getting `Kokkos::View` by reference.  That's OK for host access, but beware of capturing references to `Kokkos::View` in device lambdas for device kernels.

Reply:
@mhoemmen 
1. I'm adding an execution space fence before getLocalMatrix().
2. The references were just for the sake of readability. This section always runs in host serial, and the references never get passed anywhere else, so I think it'll be OK.